apps/assistants/sync.py

```diff
@@ -18,38 +47,118 @@ def push_assistant_to_openai(assistant: OpenAiAssistant):
         assistant.save()
 
 
+@wrap_openai_errors
+def push_file_to_openai(assistant: OpenAiAssistant, file: File):
+    client = assistant.llm_provider.get_llm_service().get_raw_client()
+    with file.file.open("rb") as fh:
+        bytesio = BytesIO(fh.read())
+    openai_file = client.files.create(
+        file=(file.name, bytesio),
+        purpose="assistants",
+    )
+    file.external_id = openai_file.id
+    file.external_source = "openai"
+    file.save()
+
+
+@wrap_openai_errors
+def delete_file_from_openai(assistant: OpenAiAssistant, file: File):
+    if not file.external_id or file.external_source != "openai":
+        return
+
+    client = assistant.llm_provider.get_llm_service().get_raw_client()
+    client.beta.assistants.files.delete(assistant_id=assistant.assistant_id, file_id=file.external_id)
+    try:
+        client.files.delete(file.external_id)
+    except openai.NotFoundError:
+        pass
+    file.external_id = ""
```

Comment:
Is is necessary to clear these fields out, since the record have already been deleted by this time?

Reply:
It's not strictly necessary since this only happens during deletion but I left it in since this function doesn't actually do the deletion so it seemed logical to reset the fields. This function could get used in other places where we keep the local file.